HTTP calls returning an int result in a class cast exception

For example, an http get which responds with just an int or a string throws a class cast exception.  That's perfectly valid json so we should support it.

```
Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.util.Map
	at com.twosigma.webtau.http.Http.createBodyDataNode(Http.java:564)
	at com.twosigma.webtau.http.Http.validateAndRecord(Http.java:494)
	at com.twosigma.webtau.http.Http.lambda$createHttpStep$3(Http.java:451)
	... 57 more
```

--- a/webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy
+++ b/webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy
@@ -1167,7 +1167,7 @@ class HttpGroovyTest implements HttpConfiguration {
         }
 
         ret.should == 123
-        ret.getClass() == Integer
+        ret.getClass().should == Integer
     }
 
     @Override

--- a/webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy
+++ b/webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy
@@ -1159,6 +1159,17 @@ class HttpGroovyTest implements HttpConfiguration {
         }
     }
 
+    @Test
+    void "handles integer json responses"() {
+        def ret = http.get('/integer') {
+            body.should == 123
+            return body
+        }
+
+        ret.should == 123
+        ret.getClass() == Integer
+    }
+
     @Override
     String fullUrl(String url) {
         if (UrlUtils.isFull(url)) {

--- a/webtau-http/src/main/java/com/twosigma/webtau/http/Http.java
+++ b/webtau-http/src/main/java/com/twosigma/webtau/http/Http.java
@@ -558,11 +558,8 @@ public class Http {
             }
 
             if (response.isJson()) {
-                Object mapOrList = JsonUtils.deserialize(response.getTextContent());
-
-                return mapOrList instanceof List ?
-                        DataNodeBuilder.fromList(id, (List<Object>) mapOrList) :
-                        DataNodeBuilder.fromMap(id, (Map<String, Object>) mapOrList);
+                Object object = JsonUtils.deserialize(response.getTextContent());
+                return DataNodeBuilder.fromValue(id, object);
             }
 
             return new StructuredDataNode(id, new TraceableValue(response.getBinaryContent()));

--- a/webtau-http/src/main/java/com/twosigma/webtau/http/datanode/DataNodeBuilder.java
+++ b/webtau-http/src/main/java/com/twosigma/webtau/http/datanode/DataNodeBuilder.java
@@ -57,7 +57,7 @@ public class DataNodeBuilder {
     }
 
     @SuppressWarnings("unchecked")
-    private static DataNode fromValue(DataNodeId id, Object value) {
+    public static DataNode fromValue(DataNodeId id, Object value) {
         if (value instanceof Map) {
             return new StructuredDataNode(id, buildMapOfNodes(id, (Map<String, Object>)value));
         } else if (value instanceof List) {

--- a/webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java
+++ b/webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java
@@ -270,6 +270,18 @@ public class HttpJavaTest implements HttpConfiguration  {
         });
     }
 
+    @Test
+    public void handlesIntegerJsonResponses() {
+        Integer ret = http.get("/integer", (header, body) -> {
+            body.should(equal(123));
+            return body;
+        });
+
+        actual(ret).should(equal(123));
+        actual(ret.getClass()).should(equal(Integer.class));
+    }
+
+
     @Override
     public String fullUrl(String url) {
         if (UrlUtils.isFull(url)) {

--- a/webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java
+++ b/webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java
@@ -281,7 +281,6 @@ public class HttpJavaTest implements HttpConfiguration  {
         actual(ret.getClass()).should(equal(Integer.class));
     }
 
-
     @Override
     public String fullUrl(String url) {
         if (UrlUtils.isFull(url)) {

--- a/webtau-http/src/test/java/com/twosigma/webtau/http/HttpTestDataServer.java
+++ b/webtau-http/src/test/java/com/twosigma/webtau/http/HttpTestDataServer.java
@@ -68,6 +68,7 @@ public class HttpTestDataServer {
         testServer.registerDelete("/resource", new TestServerTextResponse("abc"));
         testServer.registerGet("/params?a=1&b=text", new TestServerJsonResponse("{\"a\": 1, \"b\": \"text\"}"));
         testServer.registerPost("/params?a=1&b=text", new TestServerJsonResponse("{\"a\": 1, \"b\": \"text\"}", 201));
+        testServer.registerGet("/integer", new TestServerJsonResponse("123"));
 
         registerRedirects();
     }

